pennylane_lightning/core/src/utils/UtilLinearAlg.hpp

```diff
@@ -14,33 +14,72 @@
 // limitations under the License.
 
 /**
- * [@file](https://github.com/file)
+ * @file
  * LAPACK wrapper functions declarations.
  */
 #pragma once
 
 #include <algorithm>
+#include <array>
 #include <complex>
+#include <cstdlib>
+#include <filesystem>
+#include <memory>
+#include <string>
 #include <vector>
 
+#include <iostream>
+
+#include "SharedLibLoader.hpp"
+
+#include "config.h"
+
 /// @cond DEV
 namespace {
-extern "C" {
-// LAPACK routine for complex Hermitian eigensystems
-extern void cheev_(const char *jobz, const char *uplo, const int *n,
-                   std::complex<float> *a, const int *lda, float *w,
-                   std::complex<float> *work, const int *lwork, float *rwork,
-                   int *info);
-extern void zheev_(const char *jobz, const char *uplo, const int *n,
-                   std::complex<double> *a, const int *lda, double *w,
-                   std::complex<double> *work, const int *lwork, double *rwork,
-                   int *info);
-}
+// Declare heev function pointers to access corresponding functions in
+// LAPACK/OpenBLAS
+using zheevPtr = void (*)(const char *, const char *, const int *,
+                          std::complex<double> *, const int *, double *,
+                          std::complex<double> *, const int *, double *, int *);
+using cheevPtr = void (*)(const char *, const char *, const int *,
+                          std::complex<float> *, const int *, float *,
+                          std::complex<float> *, const int *, float *, int *);
+
+// Priority table used to sort openblas and its dependencies
+std::array<std::string, 5> priority_lib{"stdc", "gcc.", "quadmath", "gfortran",
+                                        "openblas"};
+
 } // namespace
 /// @endcond
 
 namespace Pennylane::Util {
 
+// Exclusively for python calls and tested in the python layer
+// LCOV_EXCL_START
+#ifdef __linux__
+/**
+ * @brief Get the path to the current shared library object.
+ *
+ * @return const char*
+ */
+inline const char *getPath() {
```

Comment:
You may want to move this logic to `SharedLibLoader.hpp`

Reply:
Let's leave it here since this is only valid for certain scenario.